include/xtensor/xnpy.hpp

```diff
@@ -42,17 +43,14 @@ namespace xt
     namespace detail
     {
 
-    /* Compile-time test for byte order.
-       If your compiler does not define these per default, you may want to define
-       one of these constants manually.
-       Defaults to little endian order. */
-#if defined(__BYTE_ORDER) && __BYTE_ORDER == __BIG_ENDIAN || defined(__BIG_ENDIAN__) || \
-    defined(__ARMEB__) || defined(__THUMBEB__) || defined(__AARCH64EB__) ||             \
-    defined(_MIBSEB) || defined(__MIBSEB) || defined(__MIBSEB__)
-        const bool big_endian = true;
-#else
-        const bool big_endian = false;
-#endif
+    	/* Test for endianess. Compiler can optimize that to a single constant. */
+	static inline bool is_big_endian() {
```

Comment:
What's the reason for declaring the method static?
Nitpicking:
- opening curly brace should be on a new line
- indentation is 4 spaces

Reply:
I declared it static inline so that there's no reason the compiler emit a symbol for it, if it can inline it, which is fine because it doesn't belong to the official API (it's in the ``details`` namespace).

Reply:
Isn't inline enough to avoid the compiler to emit a symbol?